### DISCUSSION TITLE
initial state for list_reduce

### DISCUSSION THREAD

**Discussion by soerenwolfers** [duckdb/duckdb#13455] [Ideas]
the ability to compute `f(...f(f(y0, x0), x1),..., xn)` is a 
recurring request, e.g. at https://github.com/duckdb/duckdb/issues/13397
and is particularly important and natural when the domain and codomain of `f` differ, such that this cannot be expressed as a `list_reduce(list_prepend(x, y), f)`

**Comment by maiadegraaf** [comment]
Implemented here: https://github.com/duckdb/duckdb/pull/16602
